tasks/omnibus.py

```diff
@@ -387,10 +388,51 @@ def manifest(
     )
 
 
+def _replace_dylib_path_with_rpath(ctx, dylib_paths, install_path, file):
+    for otool_line in dylib_paths.splitlines():
+        if "name" not in otool_line:
+            continue
+        dylib_path = otool_line.strip().split(" ")[1]
+        if install_path not in dylib_path:
+            continue
+        new_dylib_path = dylib_path.replace(install_path, "@rpath")
+        ctx.run(f"install_name_tool -change {dylib_path} {new_dylib_path} {file}")
+
+
+def _replace_dylib_id_path_with_rpath(ctx, dylib_paths, install_path, file):
+    for otool_line in dylib_paths.splitlines():
+        if "name" not in otool_line:
+            continue
+        dylib_path = otool_line.strip().split(" ")[1]
+        if install_path not in dylib_path:
+            continue
+        new_dylib_path = dylib_path.replace(install_path, "@rpath")
+        ctx.run(f"install_name_tool -id {new_dylib_path} {file}")
```

Comment:
Some ideas here, let me know what you think:

```suggestion
def _otool_install_path_replacements(otool_output, install_path):
    """Returns a mapping of path replacements from `otool -l` output
    where references to `install_path` are replaced by `@rpath`."""
    for otool_line in dylib_paths.splitlines():
        if "name" not in otool_line:
            continue
        dylib_path = otool_line.strip().split(" ")[1]
        if install_path not in dylib_path:
            continue
        new_dylib_path = dylib_path.replace(install_path, "@rpath")
        yield dylib_path, new_dylib_path


def _replace_dylib_paths_with_rpath(ctx, otool_output, install_path, file):
    for dylib_path, new_dylib_path in _otool_install_path_replacements(otool_output, install_path):
        ctx.run(f"install_name_tool -change {dylib_path} {new_dylib_path} {file}")


def _replace_dylib_id_path_with_rpath(ctx, otool_output, install_path, file):
    for dylib_path, new_dylib_path in _otool_install_path_replacements(otool_output, install_path):
        ctx.run(f"install_name_tool -id {new_dylib_path} {file}")
```

There might be other ways to slice it, the main point being to deduplicate common logic and making it more explicit that we're processing _otool_ output (`dylib_paths` can be somewhat misleading there).